tests/github/interactive_api_director/experiments/tensorflow_mnist/envoy/shard_descriptor.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (C) 2020-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+"""Kvasir shard descriptor."""
+
+import numpy as np
+from tensorflow import keras
+
+from openfl.interface.interactive_api.shard_descriptor import ShardDescriptor
+
+
+class MNISTShardDescriptor(ShardDescriptor):
+    """Shard descriptor class."""
+
+    def __init__(self, rank_worldsize: str = '1,1') -> None:
+        """Initialize KvasirShardDescriptor."""
+        super().__init__()
+        
+        (x_train, y_train), (x_test, y_test) = keras.datasets.mnist.load_data()
+        x_train = np.reshape(x_train, (-1, 784))
+        x_test = np.reshape(x_test, (-1, 784))
+        self.rank, self.worldsize = tuple(int(num) for num in rank_worldsize.split(','))
+
+        # Sharding
+        self.X_train = x_train[self.rank - 1::self.worldsize]
+        self.y_train = y_train[self.rank - 1::self.worldsize]
+        self.X_test = x_test[self.rank - 1::self.worldsize]
+        self.y_test = y_test[self.rank - 1::self.worldsize]
+
+
+        # Calculating data and target shapes
+        sample, target = self[0]
+        self._sample_shape = [str(dim) for dim in sample.shape]
+        self._target_shape = [str(dim) for dim in target.shape]
+
+    
+    def __getitem__(self, index):
+        """Return a item by the index."""
+        if index < len(self.X_train):
+            return self.X_train[index], self.y_train[index]
+        index -= len(self.X_train) + 1
+        return self.X_test[index], self.y_test[index]
```

Comment:
It would be greate if we have the same names for similar variables `X_train` or `train_indices` and `index` or `indice`